src/bms/command/channel.rs

```diff
@@ -175,6 +173,87 @@ pub enum PlayerSide {
     Player2,
 }
 
+/// A channel ID used in BMS files to identify channel types.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct ChannelId([u8; 2]);
+
+impl std::fmt::Debug for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("ChannelId")
+            .field(&format!("{}{}", self.0[0] as char, self.0[1] as char))
+            .finish()
+    }
+}
+
+impl std::fmt::Display for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}", self.0[0] as char, self.0[1] as char)
+    }
+}
+
+impl TryFrom<[char; 2]> for ChannelId {
+    type Error = [char; 2];
+    fn try_from(value: [char; 2]) -> core::result::Result<Self, Self::Error> {
+        Ok(Self([
+            char_to_base62(value[0]).ok_or(value)?,
+            char_to_base62(value[1]).ok_or(value)?,
+        ]))
+    }
+}
+
+impl TryFrom<[u8; 2]> for ChannelId {
+    type Error = [u8; 2];
+    fn try_from(value: [u8; 2]) -> core::result::Result<Self, Self::Error> {
+        <Self as TryFrom<[char; 2]>>::try_from([value[0] as char, value[1] as char])
+            .map_err(|_| value)
+    }
+}
+
+impl<'a> TryFrom<&'a str> for ChannelId {
+    type Error = &'a str;
+    fn try_from(value: &'a str) -> core::result::Result<Self, Self::Error> {
+        if value.len() != 2 {
+            return Err(value);
+        }
+        let mut chars = value.bytes();
+        let [Some(ch1), Some(ch2), None] = [chars.next(), chars.next(), chars.next()] else {
+            return Err(value);
+        };
+        Self::try_from([ch1, ch2]).map_err(|_| value)
+    }
+}
+
+impl From<ChannelId> for u16 {
+    fn from(value: ChannelId) -> Self {
+        base62_to_byte(value.0[0]) as u16 * 62 + base62_to_byte(value.0[1]) as u16
+    }
+}
+
+impl From<ChannelId> for u32 {
+    fn from(value: ChannelId) -> Self {
+        Into::<u16>::into(value) as u32
+    }
+}
+
+impl From<ChannelId> for u64 {
+    fn from(value: ChannelId) -> Self {
+        Into::<u16>::into(value) as u64
+    }
+}
+
+impl ChannelId {
+    /// Instances a special null id, which means the rest object.
+    pub const fn null() -> Self {
+        Self([0, 0])
+    }
+
```

Comment:
According to [<CHANNEL> in BMS command memo](https://hitkey.nekokan.dyndns.info/cmds.htm#CHANNEL), the channel `00` is a retired number. It doesn't mean rest or such special things.